griptape/config/azure_openai_structure_config.py

```diff
@@ -0,0 +1,102 @@
+from typing import Callable, Optional
+from attrs import Factory, define, field
+
+from griptape.config import StructureConfig
+from griptape.drivers import (
+    LocalVectorStoreDriver,
+    AzureOpenAiChatPromptDriver,
+    AzureOpenAiEmbeddingDriver,
+    AzureOpenAiImageGenerationDriver,
+    AzureOpenAiVisionImageQueryDriver,
+    BasePromptDriver,
+    BaseEmbeddingDriver,
+    BaseImageGenerationDriver,
+    BaseImageQueryDriver,
+    BaseVectorStoreDriver,
+)
+
+
+@define
+class AzureOpenAiStructureConfig(StructureConfig):
+    """Azure OpenAI Structure Configuration.
+
+    Attributes:
+        azure_endpoint: The endpoint for the Azure OpenAI instance.
+        azure_ad_token: An optional Azure Active Directory token.
+        azure_ad_token_provider: An optional Azure Active Directory token provider.
+        api_key: An optional Azure API key.
+        prompt_driver: An Azure OpenAI Chat Prompt Driver.
+        image_generation_driver: An Azure OpenAI Image Generation Driver.
+        image_query_driver: An Azure OpenAI Vision Image Query Driver.
+        embedding_driver: An Azure OpenAI Embedding Driver.
+        vector_store_driver: A Local Vector Store Driver.
+    """
+
+    azure_endpoint: str = field(kw_only=True, metadata={"serializable": True})
```

Comment:
Missing `azure_deployment`?

Reply:
the deployments now default to the `model`. it doesn't make sense on the top-level of the config because the deployments are per model.